File: Week_03/70/climbing_stairs.py
class Solution:
    cache = {}
    def climbStairs(self, n: int) -> int:
        if n in self.cache:
            return self.cache[n]
        if n < 3:
            v = n
        else:
            v = self.climbStairs(n-2) + self.climbStairs(n-1)
        self.cache[n] = v
        return v

File: Week_03/70/climbing_stairs.go
package main

var cache = map[int]int{}

func climbStairs(n int) int {
	if v, ok := cache[n]; ok {
		return v
	} else {
		if n < 3 {
			v = n
		} else {
			v = climbStairs(n-2) + climbStairs(n-1)
		}
		cache[n] = v
		return v
	}
}
